### PR TITLE
Add healthcheck info to CLI

### LINKED ISSUE #2415: Expose health check details in container list
In 1.12, engine adds support for container level health checks.  When querying `/containers/json` directly on the engine, this health status is bubbled up as a string appended to the Status field.  If you send this same query to swarm running on the same engine, the health information is missing.  Swarm shouldn't discard this information.

Engine example:
```
"Status": "Up 42 minutes (healthy)"
```

Swarm example for the same container:
```
"Status": "Up 42 minutes"
```

### PR DESCRIPTION
**Pull request by dongluochen**
Make CLI output consistent with Docker CLI. Fix #2415.

This change depends on engine-api vendoring.

Signed-off-by: Dong Chen <dongluo.chen@docker.com>

### CODE REVIEW

**Comment by nishanttotla** [comment]
Depends on #2422

**Comment by dongluochen** [comment]
Integration test failure is fixed in #2424.

File: cluster/container.go [hunk 1]
```diff
@@ -54,6 +54,19 @@ func FullStateString(state *types.ContainerState) string {
...
+		healthText := ""
+		if h := state.Health; h != nil {
+			switch h.Status {
+			case types.Starting:
+				healthText = "health: starting"
+			default: // Healthy and Unhealthy are clear on their own
+				healthText = h.Status
+			}
+		}
+		if len(healthText) > 0 {
```
Comment (nishanttotla) [inline cluster/container.go]: Can't `healthText` be failed or something? In that case, should we still print `Up...`?

Comment (dongluochen) [inline cluster/container.go]: In that case container is up and running. It's just failing health check. So I think it's a reasonable combination.

Comment (nishanttotla) [inline cluster/container.go]: Sorry my bad, this makes sense.

**Comment by nishanttotla** [comment]
LGTM.

Ping @abronan

**Comment by abronan** [comment]
LGTM

**Pull request merged**